### PR TITLE
fix: Use Containerfile instead s2i

### PR DESCRIPTION
**Pull request by mateusoliveira43**
- [x] I have read [Pelorus Contributing guidelines](https://github.com/dora-metrics/pelorus/blob/master/CONTRIBUTING.md)

## Linked Issues

related to #862

## Description

Use Containerfile instead of s2i to create exporters images.

### Test in OpenShift

Create Pelorus instance with
```yaml
kind: Pelorus
apiVersion: charts.pelorus.dora-metrics.io/v1alpha1
metadata:
  name: test-containerfile
  namespace: test-pelorus-operator
spec:
  exporters:
    global: {}
    instances:
      - app_name: committime-exporter
        exporter_type: committime
        source_url: https://github.com/mateusoliveira43/pelorus
        source_ref: fix/use-containerfile-instead-s2i
```
using operator created by CI commented in this PR.

Check the logs from exporter build Pod and run `skopeo --help` in the terminal of running exporter Pod.

### Tests with E2E 

Check if e2e tests used Containerfile to create images.

For example, run `./scripts/run-pelorus-e2e-tests.sh -o konveyor` and check the logs from exporters build Pod and run `skopeo --help` in the terminal of running exporters Pod.

### After PR closes

Check if CI of following PRs work as expected:
- test images are created successfully
- exporters images are created successfully

### CODE REVIEW

File: container/Containerfile
Comment (mateusoliveira43) [inline container/Containerfile:1]: add this to docs

File: docs/Development.md
Comment (mateusoliveira43) [inline docs/Development.md:256]: TODO

File: docs/Development.md
Comment (mateusoliveira43) [inline docs/Development.md:361]: TODO

**Comment by mateusoliveira43** [comment]
/hold 
discuss before merging

File: .github/workflows/build-pelorus.yml
Comment (mateusoliveira43) [inline .github/workflows/build-pelorus.yml:4]: TODO

File: container/Containerfile
Comment (mateusoliveira43) [inline container/Containerfile:27]: TODO

File: exporters/Containerfile
Comment (mateusoliveira43) [inline exporters/Containerfile:1]: use fixed TAG?

File: exporters/Containerfile
Comment (mateusoliveira43) [inline exporters/Containerfile:17]: Do we need to clean cache in this layer? In debian images I usually run `rm -rf /var/lib/apt/lists/*` after `apt-get install`

Comment (mpryc) [reply exporters/Containerfile:17]: yes, any cleanup done in this step will reduce it's size. But it needs to be within line 17.

File: exporters/Containerfile
Comment (mateusoliveira43) [reply exporters/Containerfile:1]: did not like the available options, going to stick with latest
Screenshot from 2023-05-31 16-58-47

File: .github/workflows/build-pelorus.yml
Comment (mateusoliveira43) [inline .github/workflows/build-pelorus.yml:46]: TODO

File: .github/workflow_templates/build_push_exporters/action.yml
Comment (mateusoliveira43) [inline .github/workflow_templates/build_push_exporters/action.yml:40]: this will break in test images job :(

**Review by mpryc — COMMENTED** [review]
/lgtm

**Review by mpryc — APPROVED** [review]

**Pull request merged**
